src/helpers.ts

```diff
@@ -516,10 +530,16 @@ export function hexToBuffer(hex: string): Buffer {
     return Buffer.alloc(0);
   }
   if (!hex.startsWith('0x')) {
-    throw new Error(`Hex string is missing the "0x" prefix: "${hex}"`);
+    throw new InvalidRequestError(
```

Comment:
There's a bit of a separation of concern issue here. We're adding in public API Express route handling code into this `hexToBuffer` function, which is used in a lot of areas unrelated to the express routes.

The alternatives are more LOC, but would be a lot more explicit and we'd probably better off. One alternative could be going through the express routes, and validating any request params. Something we should probably be doing anyway. An example would be something like:
```ts
router.getAsync('/:tx_id', async (req, res, next) => {
  const { tx_id } = req.params;
  if (!has0xPrefix(tx_id)) {
    return res.redirect('/extended/v1/tx/0x' + tx_id);
  }

  try {
    hexToBuffer(tx_id);
  } catch (error: any) {
    throw new InvalidRequestError(error.message, InvalidRequestErrorType.invalid_tx_id);
  }

  ...

});
```

It should be pretty straightforward to do this for all the routes that take a hex-string input.